### PR TITLE
CI-34: Login

### PR DESCRIPTION
**Pull request by SerhiiBorozenets**
[CI-34](https://cloverpop-internship.atlassian.net/browse/CI-34)

[CI-34]: https://cloverpop-internship.atlassian.net/browse/CI-34?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -1,5 +1,13 @@
 class HomeController < ApplicationController
-  def index; end
+  include ApplicationHelper
+  before_action :require_user!, only: :app
+
+  def index
+    return redirect_to app_path if current_user.present?
+    return redirect_to auth.sign_in_path, flash: { error: 'Expired!' } if session_expired?
```
Comment (rogergraves) [inline app/controllers/home_controller.rb:7]: Let's make the flash message more informative: "The last login link you tried using expired. Please request a new one."

Comment (SerhiiBorozenets) [reply app/controllers/home_controller.rb:7]: @rogergraves I don't show this flash for users and use it only for condition in new.html.erb
`<% if flash.to_h["error"] == 'Expired!' %>
      <h3 class="muted my-4">Your login link may have expired. Click here to get a new one</h3>
    <% else %>`

File: app/admin/passwordless_sessions.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
+ActiveAdmin.register Passwordless::Session do
```
[the remark this answers is not in the document]
Comment (SerhiiBorozenets) [reply app/admin/passwordless_sessions.rb:1]: @rogergraves in [CI-38](https://cloverpop-internship.atlassian.net/browse/CI-38) we have requirements for AA: `Please add the ability to update “Login link expire” date and time on Edit user page in Active Admin`

If you think that the information on sessions in AA is redundant, I can leave the editing  `expire` only.

[the remark this answers is not in the document]
Comment (rogergraves) [reply app/admin/passwordless_sessions.rb:1]: @SerhiiBorozenets sounds like a good plan. Please hide the actual token string from all actions and allow the editing of the expiring date as you suggested.

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -1,2 +1,21 @@
...
+  private
+
+  def current_user
+    @current_user ||= authenticate_by_session(User)
+  end
+
+  def require_user!
+    return if current_user
+
+    redirect_to root_path, flash: { error: 'You are not worthy!' }
```
[the remark this answers is not in the document]
Comment (SerhiiBorozenets) [reply app/helpers/application_helper.rb:13]: @rogergraves maybe "Sorry, you should try logging in again!"

**Review by rogergraves — APPROVED** [review]

**Pull request merged**